api/v1alpha1/siteconfig_types.go

```diff
@@ -0,0 +1,461 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	bmh_v1alpha1 "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+
+	aiv1beta1 "github.com/openshift/assisted-service/api/v1beta1"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// SshPrivateKeySecretRef defines the private key secret reference
+type SshPrivateKeySecretRef struct {
+	Name string `json:"name,omitempty"`
+}
+
+// MachineNetworkEntry is a single IP address block for node IP blocks.
+type MachineNetworkEntry struct {
+	// CIDR is the IP block address pool for machines within the cluster.
+	CIDR string `json:"cidr"`
+}
+
+// ClusterNetworkEntry is a single IP address block for pod IP blocks. IP blocks
+// are allocated with size 2^HostSubnetLength.
+type ClusterNetworkEntry struct {
+	// CIDR is the IP block address pool.
+	CIDR string `json:"cidr"`
+
+	// HostPrefix is the prefix size to allocate to each node from the CIDR.
+	// For example, 24 would allocate 2^8=256 adresses to each node. If this
+	// field is not used by the plugin, it can be left unset.
+	// +optional
+	HostPrefix int32 `json:"hostPrefix,omitempty"`
+}
+
+// BmcCredentialsName
+type BmcCredentialsName struct {
+	Name string `json:"name,omitempty"`
+}
+
+// IronicInspect
+type IronicInspect string
+
+// SiteConfigMap
+type SiteConfigMap struct {
+	Name string `json:"name,omitempty"`
+	// +kubebuilder:default:=ztp-site
+	Namespace string            `json:"namespace,omitempty"`
+	Data      map[string]string `json:"data,omitempty"`
+}
+
+type CrAnnotations struct {
```

Comment:
Is there a better name we can give this than `Cr`? Maybe just `Annotations`? Or `RenderedResourceAnnotations` (a bit long :worried: )?

I don't want to cause huge breaks in compatibility, but while we're doing a v2 maybe we can think about stuff like this...

Reply:
`Annotations` sounds good to me. @imiller0 can you chime in on this (@carbonin makes a valid point w.r.t. improving the SiteConfig)?

Reply:
Changed to `extraAnnotations`